syntax = "proto3";

service TimeSeriesService {
  rpc insert (TimeSeries) returns (ReindexingResponse) {}
  rpc query (TimeSeries) returns (TimeSeriesIdx) {}
  rpc reindex (ReindexingRequest) returns (ReindexingResponse) {}
  rpc save(SaveIndexRequest) returns (SaveIndexResponse) {}
  rpc load(LoadIndexRequest) returns (LoadIndexResponse) {}
}

message SaveIndexRequest {
  string name = 1;
}

message SaveIndexResponse {
  int32 response = 1;
}

message LoadIndexRequest {
  string name = 1;
}

message LoadIndexResponse {
  int32 response = 1;
}

message ReindexingResponse {
  int32 status = 1;
  int32 ts_id = 2;
}

message ReindexingRequest {
  int32 n_samples = 1;
}

message TimeSeries {
  repeated float ts = 1;
  int32 length = 2;
  int32 dim = 3;
}

message TimeSeriesIdx {
  repeated int32 ids = 1;
  int32 status = 2;
}